### DISCUSSION TITLE
Build fails: cmd/exec.go:41:9: undefined: run

### DISCUSSION THREAD

**Discussion by yurivict** [knqyf263/pet#292] [General]
```
===>  Building for pet-0.6.0
(cd /usr/ports/deskutils/pet/work/github.com/knqyf263/pet@v0.6.0;  for t in pet; do  out=$(/usr/bin/basename $(echo ${t} |  /usr/bin/sed -Ee 's/^[^:]*:([^:]+).*$/\1/' -e 's/^\.$/pet/'));  pkg=$(echo ${t} |  /usr/bin/sed -Ee 's/^([^:]*).*$/\1/' -e 's/^pet$/./');  echo "===>  Building ${out} from ${pkg}";  /usr/bin/env -i HOME=/usr/ports/deskutils/pet/work  PWD="${PWD}" OSVERSION=1400502 PATH=/usr/local/libexec/ccache:/usr/ports/deskutils/pet/work/.bin:/home/yuri/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin TERM=xterm-256color XDG_DATA_HOME=/usr/ports/deskutils/pet/work  XDG_CONFIG_HOME=/usr/ports/deskutils/pet/work  XDG_CACHE_HOME=/usr/ports/deskutils/pet/work/.cache  HOME=/usr/ports/deskutils/pet/work PATH=/usr/local/libexec/ccache:/usr/ports/deskutils/pet/work/.bin:/home/yuri/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin PKG_CONFIG_LIBDIR=/usr/ports/deskutils/pet/work/.pkgconfig:/usr/local/libdata/pkgconfig:/usr/local/share/pkgconfig:/usr/libdata/pkgconfig MK_DEBUG_FILES=no MK_KERNEL_SYMBOLS=no SHELL=/bin/sh NO_LINT=YES PREFIX=/usr/local  LOCALBASE=/usr/local  CC="cc" CFLAGS="-O2 -pipe  -fstack-protector-strong -fno-strict-aliasing "  CPP="cpp" CPPFLAGS=""  LDFLAGS=" -fstack-protector-strong " LIBS=""  CXX="c++" CXXFLAGS="-O2 -pipe -fstack-protector-strong -fno-strict-aliasing  " CCACHE_DIR="/tmp/.ccache" BSD_INSTALL_PROGRAM="install  -s -m 555"  BSD_INSTALL_LIB="install  -s -m 0644"  BSD_INSTALL_SCRIPT="install  -m 555"  BSD_INSTALL_DATA="install  -m 0644"  BSD_INSTALL_MAN="install  -m 444" CGO_ENABLED=1  CGO_CFLAGS="-I/usr/local/include"  CGO_LDFLAGS="-L/usr/local/lib"  GOAMD64=  GOARM=  GOTMPDIR="/usr/ports/deskutils/pet/work" GOPATH="/usr/ports/distfiles/go/deskutils_pet"  GOBIN="/usr/ports/deskutils/pet/work/bin"  GO111MODULE=on  GOFLAGS=-modcacherw  GOSUMDB=sum.golang.org GOMAXPROCS=7 GOPROXY=off /usr/local/bin/go122 build -buildmode=exe -v -trimpath -ldflags=-s -buildvcs=false -mod=vendor  -o /usr/ports/deskutils/pet/work/bin/${out}  ${pkg};  done)
===>  Building pet from .
github.com/knqyf263/pet/cmd
# github.com/knqyf263/pet/cmd
cmd/exec.go:41:9: undefined: run
cmd/util.go:17:9: undefined: run
cmd/util.go:64:8: undefined: run
*** Error code 1
```

Version: 0.6.0
FreeBSD 14.0

**Comment by RamiAwar** [comment]
> ```
> ===>  Building for pet-0.6.0
> (cd /usr/ports/deskutils/pet/work/github.com/knqyf263/pet@v0.6.0;  for t in pet; do  out=$(/usr/bin/basename $(echo ${t} |  /usr/bin/sed -Ee 's/^[^:]*:([^:]+).*$/\1/' -e 's/^\.$/pet/'));  pkg=$(echo ${t} |  /usr/bin/sed -Ee 's/^([^:]*).*$/\1/' -e 's/^pet$/./');  echo "===>  Building ${out} from ${pkg}";  /usr/bin/env -i HOME=/usr/ports/deskutils/pet/work  PWD="${PWD}" OSVERSION=1400502 PATH=/usr/local/libexec/ccache:/usr/ports/deskutils/pet/work/.bin:/home/yuri/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin TERM=xterm-256color XDG_DATA_HOME=/usr/ports/deskutils/pet/work  XDG_CONFIG_HOME=/usr/ports/deskutils/pet/work  XDG_CACHE_HOME=/usr/ports/deskutils/pet/work/.cache  HOME=/usr/ports/deskutils/pet/work PATH=/usr/local/libexec/ccache:/usr/ports/deskutils/pet/work/.bin:/home/yuri/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin PKG_CONFIG_LIBDIR=/usr/ports/deskutils/pet/work/.pkgconfig:/usr/local/libdata/pkgconfig:/usr/local/share/pkgconfig:/usr/libdata/pkgconfig MK_DEBUG_FILES=no MK_KERNEL_SYMBOLS=no SHELL=/bin/sh NO_LINT=YES PREFIX=/usr/local  LOCALBASE=/usr/local  CC="cc" CFLAGS="-O2 -pipe  -fstack-protector-strong -fno-strict-aliasing "  CPP="cpp" CPPFLAGS=""  LDFLAGS=" -fstack-protector-strong " LIBS=""  CXX="c++" CXXFLAGS="-O2 -pipe -fstack-protector-strong -fno-strict-aliasing  " CCACHE_DIR="/tmp/.ccache" BSD_INSTALL_PROGRAM="install  -s -m 555"  BSD_INSTALL_LIB="install  -s -m 0644"  BSD_INSTALL_SCRIPT="install  -m 555"  BSD_INSTALL_DATA="install  -m 0644"  BSD_INSTALL_MAN="install  -m 444" CGO_ENABLED=1  CGO_CFLAGS="-I/usr/local/include"  CGO_LDFLAGS="-L/usr/local/lib"  GOAMD64=  GOARM=  GOTMPDIR="/usr/ports/deskutils/pet/work" GOPATH="/usr/ports/distfiles/go/deskutils_pet"  GOBIN="/usr/ports/deskutils/pet/work/bin"  GO111MODULE=on  GOFLAGS=-modcacherw  GOSUMDB=sum.golang.org GOMAXPROCS=7 GOPROXY=off /usr/local/bin/go122 build -buildmode=exe -v -trimpath -ldflags=-s -buildvcs=false -mod=vendor  -o /usr/ports/deskutils/pet/work/bin/${out}  ${pkg};  done)
> ===>  Building pet from .
> github.com/knqyf263/pet/cmd
> # github.com/knqyf263/pet/cmd
> cmd/exec.go:41:9: undefined: run
> cmd/util.go:17:9: undefined: run
> cmd/util.go:64:8: undefined: run
> *** Error code 1
> ```
> 
> 
> Version: 0.6.0
> FreeBSD 14.0

@yurivict Thanks for submitting this! Does indeed look like a lack of support for the OS, will look into it. Should be an easy fix.

**Comment by RamiAwar** [comment]
@yurivict What command did you run to build this? From what I see this `//go:build !windows` in cmd/util_unix.go should support freeBSD.

Also any specific reason why you're on v0.6.0 and not latest?

**Comment by yurivict** [comment]
@RamiAwar 

The automatic release discovery system reported 0.6.0, and failed to find 0.7.0

0.7.0 builds fine on FreeBSD.

Thank you for your help!

**Reply by RamiAwar** [reply]
Weird! Well glad it works now! What are you using for the version discovery? Maybe I should look into that 😁

**Reply by yurivict** [reply]
This is a custom program called "portscout", specific to FreeBSD.

https://portscout.freebsd.org/
